app/reducers/galleryPictures.js

```diff
@@ -13,17 +31,72 @@ export type GalleryPictureEntity = {
   description: string,
   text: string,
   active: boolean,
-  comments: Array<number>
+  comments: Array<number>,
+  file: string,
+  thumbnail: string,
+  rawFile: string
 };
 
 function mutateGalleryPicture(state: any, action: any) {
+  const { uploadStatus = initialUploadStatus } = state;
   switch (action.type) {
     case GalleryPicture.DELETE.SUCCESS: {
       return {
         ...state,
         items: state.items.filter(id => id !== action.meta.id)
       };
     }
+    case Gallery.UPLOAD.BEGIN: {
+      const imageCount = uploadStatus.imageCount + action.meta.imageCount;
+      return {
+        ...state,
+        uploadStatus: {
+          ...uploadStatus,
+          imageCount,
+          showStatus: true
+        }
+      };
+    }
+    case Gallery.HIDE_UPLOAD_STATUS: {
+      return {
+        ...state,
+        uploadStatus: {
+          ...uploadStatus,
+          showStatus: false
+        }
+      };
+    }
+    case GalleryPicture.CREATE.SUCCESS: {
+      const successCount = uploadStatus.successCount + 1;
+
+      // Only update thumbnail in upload status for each 10th picture
+      const lastUploadedImage =
+        !uploadStatus.lastUploadedImage || successCount % 10 == 0
```

Comment:
Hmm, what's the motivation behind this? Performance, I guess? It feels a bit.. unresponsive? Maybe that's just me. Having it update on every upload probably doesn't work too well if the upload is as fast as we want it to be. Maybe a progress bar is better? 

Very nitpicky of course, could easily merged as is. Looks good overall!